web3/contract.py

```diff
@@ -645,6 +665,103 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+class ContractConstructor:
+    """
+    Class for contract constructor API.
+    """
+    def __init__(self, web3, address, abi, bytecode, args, kwargs):
+        self.web3 = web3
+        self.address = address
+        self.abi = abi
+        self.bytecode = bytecode
+        self.data_in_transaction = self._encode_data_in_transaction(args, kwargs)
+
+    @combomethod
+    def _encode_data_in_transaction(self, args=None, kwargs=None):
```

Comment:
_Just an idea_

Maybe this whole method could be moved to the contract utils, as:

```py
def encode_constructor_data(web3, abi, bytecode, args, kwargs):
```

and remove it from this class and the contract class, using the utility method instead.